code/modules/detective_work/forensicmachines.dm

```diff
@@ -0,0 +1,316 @@
+		to_chat(user, "<span class='notice'>You have stopped analyzing [swab]</span>")
+		scanning = FALSE
+		update_appearance(UPDATE_ICON)
+
+	to_chat(user, "<span class='notice'>Printing report...</span>")
+	var/obj/item/paper/report = new(get_turf(src))
+	report.stamped = list(/obj/item/stamp)
+	report.overlays = list("paper_stamped")
+	report_num++
+
+	if(swab)
+		var/obj/item/forensics/swab/bloodswab = swab
+		report.name = ("DNA scanner report no.[++report_num]: [bloodswab.name]")
+		//dna data itself
+		var/data = "No analysis data available."
+		if(!isnull(bloodswab.dna))
+			data = "Spectrometric analysis on the provided sample determined the presence of DNA. DNA String(s) found: [bloodswab.dna.len].<br><br>"
+			for(var/blood in bloodswab.dna)
+				data += "<span class='notice'>Blood type: [bloodswab.dna[blood]]<br>\nDNA: [blood]<br><br></span>"
+		else
+			data += "\nNo DNA found.<br>"
+		report.info = "<b>Report number: [report_num] по \n[src]</b><br>"
+		report.info += "<b>\nAnalyzed object:</b><br>[bloodswab.name]<br>[bloodswab.desc]<br><br>" + data
+		report.forceMove(src.loc)
+		report.update_icon()
+		scanning = FALSE
+		update_appearance(UPDATE_ICON)
+
+/obj/machinery/dnaforensics/proc/remove_sample(mob/living/remover)
+	if(!istype(remover) || remover.incapacitated() || !Adjacent(remover))
+		return
+	if(!swab)
+		to_chat(remover, "<span class='warning'>There is no sample inside the scanner!</span>")
+		return
+	to_chat(remover, "<span class='notice'>you removed out [swab] from the scanner.</span>")
+	swab.forceMove(get_turf(src))
+	remover.put_in_hands(swab)
+	swab = null
+	update_appearance(UPDATE_ICON_STATE)
+
+/obj/machinery/dnaforensics/AltClick()
+	remove_sample(usr)
+
+/obj/machinery/dnaforensics/MouseDrop(atom/other)
+	if(usr == other)
+		remove_sample(usr)
+	else
+		return ..()
+
+/obj/machinery/dnaforensics/update_icon_state()
+	icon_state = "dnaopen"
+	if(swab)
+		icon_state = "dnaclosed"
+		if(scanning)
+			icon_state = "dnaworking"
+
+/obj/machinery/dnaforensics/screwdriver_act(mob/user, obj/item/I)
+	if(swab)
+		return
+	. = TRUE
+	default_deconstruction_screwdriver(user, "dnaopenunpowered", "dnaopen", I)
+
+/obj/machinery/dnaforensics/wrench_act(mob/user, obj/item/I)
+	. = TRUE
+	default_unfasten_wrench(user, I)
+
+/obj/machinery/dnaforensics/crowbar_act(mob/user, obj/item/I)
+	if(swab)
+		return
+	. = TRUE
+	default_deconstruction_crowbar(user, I)
+
+// Microscope code itself
+// This is the output of the stringpercent(print) proc, and means about 80% of
+// the print must be there for it to be complete.  (Prints are 32 digits)
+/obj/machinery/microscope
+	name = "\improper Microscope"
+	desc = "A microscope capable of magnifying images up to 3000 times."
+	icon = 'icons/obj/forensics/forensics.dmi'
+	icon_state = "microscope"
+	anchored = TRUE
+	density = TRUE
+	new_attack_chain = FALSE
+	var/obj/item/sample = null
+	var/report_num = FALSE
+	var/fingerprint_complete = 6
```

Comment:
Please indicate what this variable is doing and what it indicates. If it's a constant, consider making it a define.